pkg/util/port_test.go

```diff
@@ -0,0 +1,222 @@
+package util
+
+import (
+	"testing"
+)
+
+func TestParsePort(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expected    int
+		expectError bool
+		errorMsg    string
+	}{
+		// Plain port number tests
+		{
+			name:     "valid plain port number",
+			input:    "8288",
+			expected: 8288,
+		},
+		{
+			name:     "port 80",
+			input:    "80",
+			expected: 80,
+		},
+		{
+			name:     "port 443",
+			input:    "443",
+			expected: 443,
+		},
+		{
+			name:     "high port number",
+			input:    "65535",
+			expected: 65535,
+		},
+		{
+			name:     "port 1",
+			input:    "1",
+			expected: 1,
+		},
+		
+		// TCP URL tests
+		{
+			name:     "tcp url with ip and port",
+			input:    "tcp://192.168.194.165:8288",
+			expected: 8288,
+		},
+		{
+			name:     "tcp url with localhost",
+			input:    "tcp://localhost:3000",
+			expected: 3000,
+		},
+		{
+			name:     "tcp url with hostname",
+			input:    "tcp://example.com:443",
+			expected: 443,
+		},
+		{
+			name:     "tcp url with different port",
+			input:    "tcp://10.0.0.1:9000",
+			expected: 9000,
+		},
+		{
+			name:     "tcp url with ipv6",
+			input:    "tcp://[::1]:8080",
+			expected: 8080,
+		},
+		
+		// HTTP/HTTPS URL tests (should work with any scheme)
+		{
+			name:     "http url",
+			input:    "http://localhost:8000",
+			expected: 8000,
+		},
+		{
+			name:     "https url",
+			input:    "https://example.com:8443",
+			expected: 8443,
+		},
+		
+		// Error cases
+		{
+			name:        "empty string",
+			input:       "",
+			expectError: true,
+			errorMsg:    "port cannot be empty",
+		},
+		{
+			name:        "invalid port number",
+			input:       "invalid",
+			expectError: true,
+			errorMsg:    "invalid port \"invalid\"",
+		},
+		{
+			name:     "negative port number",
+			input:    "-123",
+			expected: -123, // strconv.Atoi actually allows negative numbers
+		},
+		{
+			name:        "port number too large",
+			input:       "99999",
+			expected:    99999, // strconv.Atoi allows this, validation would be separate
+		},
+		{
+			name:        "float as port",
+			input:       "8080.5",
+			expectError: true,
+			errorMsg:    "invalid port \"8080.5\"",
+		},
```

Comment:
what happens if we have these ports at the end of a URL, does `url.Parse` return an error?